### ISSUE TITLE
mlp.max_pool returns -inf results when pool_stride > pool_shape

### ISSUE THREAD

**Issue by JesseLivezey** [lisa-lab/pylearn2#1382]
This code
```
from pylearn2.models.mlp import max_pool 
import theano 
from theano import tensor 
import numpy as np 

X = tensor.tensor4('X') 
pool = max_pool(X, pool_shape=(2,2), pool_stride=(4,4), image_shape=(4,4)) 
f = theano.function([X], outputs=pool) 

t = np.ones((1,1,4,4)).astype('float32') 
print f(t)
```
results in
```
[[[[ 1. -inf] 
[-inf -inf]]]]
```

Also posted here
https://groups.google.com/forum/#!topic/pylearn-dev/I8miDXGrAwQ

The current pooling logic can have pool locations in the larger allocated tensor that don't have any real inputs.

**Comment by goodfeli** [comment]
The symbolic max pooling code is incredibly slow and should probably just be euthanized now that we have actual cuda-based max pooling.

**Comment by nouiz** [comment]
Theano have pooling for the GPU for some time already. But not with strides
support. We have wrapped cudnn pooling that support more option like
strides, maybe this one would be good in that case. But the current
wrapping do not have the parameter border_mode=ignore. (Or the other way
around). So the border handling isn't the one pylearn2 use currently.

I think finishing the cudnn wrapping would allow to use it directly in
pylearn2 when it is available.

Until then (or in case cudnn isn't available), I think that we need to keep
the old code. Can you make a PR that make it raise an error is stride >
patch size? At least, there won't be bad value returned.

Fred

**Comment by JesseLivezey** [comment]
I have a PR: #1361, in which I'm trying to both add cudnn support when available and change the behavior when strides > shape.

Feedback would be helpful.

**Comment by JesseLivezey** [comment]
Closed by #1361.
